### DISCUSSION TITLE
Connecting TSL2591 causes Freeze/Reset of ESP32

### DISCUSSION THREAD

**Discussion by Farmboy94** [arendst/Tasmota#20520] [Q&A] [answered]
Hi there, I am currently trying to get a TSL2591 Light sensor functioning with an ESP32 and Tasmota on the most recent version. For some reason when the i2c pins are connected and I save a settings change or refresh the page it goes offline, and occasionally wipes the ESP32.  It never does recognize that it was a TSL2591 connected but sometimes it shows that a VL53L0X is connected.   When I could get into the logs it never showed any errors, i2c scan would occasionally recognize that something was on the correct address.  It did that with 2 different boards, 2 different sensors, same results.  I also hooked up a SHT31 sensor and that functioned as expected.  I attempted to remove the driver for the aforementioned VL53L0X using I2cDriver45 0" at the suggestion of another user, but the only thing I could still get to show up occasionally was the VL53L0X.  The kicker is, I flashed it with an example program from IDE and it functions perfectly fine.  Same pins being used and everything...  Has anyone ever ran into something like this before?

**Comment by barbudor** [comment]
Have you checked https://tasmota.github.io/docs/BUILDS/#available-features-and-sensors to see if TSL2591 driver was included in any standard build? 
It is not so you have to compile you own binary with
#define USE_TSL2591

Then you can check with 'i2cdriver' that driver is present and enabled
Preferably disable all other sensors that you don't use to avoid conflicts on i2c address

**Reply by Farmboy94** [reply]
It apparently isn't.  I thought I had seen it on another supported device list on the tasmota page.  So is this normal behavior when a driver for a sensor isn't included?

**Reply by barbudor** [reply, marked as answer]
VL53x is likely included and enabled
I2C generally doesn't allow to distinguish parts apart from the address
So the VL53x driver see an response from address 0x29 and has no way to find out it's not the right chip
That leads to a crash for some reason

You should disable driver 31 to avoid the bad detection
And have a build with TSL (driver 40)

**Reply by sfromis** [reply]
> I attempted to remove the driver for the aforementioned VL53L0X using I2cDriver45 0" at the suggestion of another user

That command was not what I suggested, as I specifically suggested you to use the **Index** column for the driver number. And index 45 is HDC1080, while VL53L0X is I2CDriver index 31.

**Reply by Farmboy94** [reply]
Ok, I will give that a try! Sorry @sfromis, I completely missed the numbers on the left side somehow!  I didn't come over here because I was dissatisfied, with the responses on discord.  I was just having a hard time following the conversation when its jumping from one thing to the next in the server lol.

**Reply by Farmboy94** [reply]
Compiled a version with the TSL2591, but initially forgot to disable the driver for the VL53L0X and had the same problem.  Realized my mistake and disabled the driver and now it works!

**Reply by sfromis** [reply]
It is also possible to build with I2C drivers disabled.
https://github.com/arendst/Tasmota/blob/development/tasmota/my_user_config.h#L585-L589
```
// -- I2C sensors ---------------------------------
#define USE_I2C                                  // I2C using library wire (+10k code, 0k2 mem, 124 iram)
#define I2CDRIVERS_0_31        0xFFFFFFFF          // Enable I2CDriver0  to I2CDriver31
#define I2CDRIVERS_32_63       0xFFFFFFFF          // Enable I2CDriver32 to I2CDriver63
#define I2CDRIVERS_64_95       0xFFFFFFFF          // Enable I2CDriver64 to I2CDriver95
```

What I usually do is to just have all zeros in the bits, meaning that I start with all drivers disabled, and then I enable specific ones I need after first boot.
